### PR TITLE
Fix doc nits

### PR DESCRIPTION
**Pull request by bitfield**
Clippy is wonderful, and reading its lints is a great way to learn about Rust. While doing this, I noticed a few little copyedits, such as adding trailing periods to sentences, or tweaking wording very slightly to improve its readability. I hope you will accept these suggestions as an act of love for the project, with my thanks for all the maintainers' hard work.

changelog: Docs [ mut_range_bound ]: fix doc nits
changelog: Docs [ needless_for_each ]: fix doc nits
changelog: Docs [ arc_with_non_send_sync ]: fix doc nits
changelog: Docs [ allow_attributes ]: fix doc nits
changelog: Docs [ allow_attributes_without_reason ]: fix doc nits

### CODE REVIEW

File: clippy_lints/src/needless_for_each.rs [hunk 1]
```diff
@@ -31,8 +31,8 @@ declare_clippy_lint! {
     /// Use instead:
     /// ```no_run
     /// let v = vec![0, 1, 2];
-    /// for elem in v.iter() {
-    ///     println!("{}", elem);
+    /// for elem in v {
+    ///     println!("{elem}");
```
[the remark this answers is not in the document]
Comment (bitfield) [reply clippy_lints/src/needless_for_each.rs]: Thanks, @Alexendoo! Good catch, changed as suggested.

**Review by Alexendoo — APPROVED** [review]

**Comment by Alexendoo** [comment]
Thanks!

@bors r+

**Comment by bitfield** [comment]
@Alexendoo thanks for your guidance on this one! I have a few more textual nits I've been collecting—from your point of view, what would be the best way to submit these? 

1. A single PR with lots of commits, one per lint
2. A single PR with a single commit fixing lots of lints at once
3. Multiple PRs, each with a small batch of fixes
4. Something else

Smaller PRs are faster to review, but on the other hand batching a lot of changes together reduces the admin overhead of approval and merging—as a maintainer, which do you prefer?

**Comment by Alexendoo** [comment]
I prefer single commits usually, single or multiple PRs are fine either way

I'd say don't split it up into multiple PRs if you have the work done already, but also you don't have to wait until you have done everything before opening a PR. It being in several chunks is perfectly fine if it works out like that

Whichever you go for, thank you for the contributions!

**Pull request merged**